internal/pkg/service/stream/storage/node/coordinator/fileimport/operator.go

```diff
@@ -304,22 +314,38 @@ func (o *operator) importFile(ctx context.Context, file *fileData) {
 	if err == nil {
 		o.logger.Info(ctx, "imported file")
 	}
+
+	finalizationCtx := context.WithoutCancel(ctx)
+
+	// Update telemetry
+	attrs := append(
+		file.FileKey.SinkKey.Telemetry(), // Anything more specific than SinkKey would make the metric too expensive
+		attribute.String("error_type", telemetry.ErrorType(err)),
+		attribute.String("operation", "fileimport"),
+	)
+	durationMs := float64(o.clock.Now().Sub(startTime)) / float64(time.Millisecond)
+	o.metrics.Duration.Record(finalizationCtx, durationMs, metric.WithAttributes(attrs...))
+	o.metrics.Compressed.Add(finalizationCtx, int64(stats.CompressedSize), metric.WithAttributes(attrs...))
+	o.metrics.Uncompressed.Add(finalizationCtx, int64(stats.UncompressedSize), metric.WithAttributes(attrs...))
```

Comment:
Yes and no:
- duration :heavy_check_mark: 
- size :x: - because the data was not uploaded/imported
  - It can be filtered out in DD, but I assume that someone will surely forget about it one day.
  - Do metrics about the size of failed operations have an informational value for us?

Reply:
There is a difference in the router, because the request with the given size arrived, `but ended with an error - there is no retry on the server side.